tests/foreman/ui/test_ansible.py

```diff
@@ -519,6 +519,55 @@ def test_positive_assign_and_remove_ansible_role_to_hostgroup(
             session.hostgroup.delete(name)
             assert not target_sat.api.HostGroup().search(query={'search': f'name={name}'})
 
+    @pytest.mark.tier3
+    def test_positive_non_admin_hostgroup_permission(
+        self, request, function_org, function_location, target_sat
+    ):
+        """Verify Hostgroup page is accessible to non-admin user.
+
+        :id: 3b8ce895-9d4c-491d-a5eb-d292748134a5
+
+        :customerscenario: true
+
+        :BZ: 2236418
+
+        :steps:
+            1. Create a user with non-admin permissions and assign "Ansible Roles Manager", "Ansible Tower Inventory Reader"
+            and "Content Importer" to the user.
+            2. Create a hostgroup and add in same organization and location as user.
+            3. Login to Satellite WebUI with user credentials and navigate to Configure -> Hostgroups
+
+        :expectedresults: The hostgroup created should be visible to non-admin user.
+        """
+        password = settings.server.admin_password
+
+        ansible_manager_role = (
+            target_sat.api.Role().search(query={'search': 'name="Ansible Roles Manager"'})[0].id
+        )
+        tower_inventory_manager_role = (
+            target_sat.api.Role()
+            .search(query={'search': 'name="Ansible Tower Inventory Reader"'})[0]
+            .id
+        )
+        SELECTED_ROLE = [ansible_manager_role, tower_inventory_manager_role]
+        user = target_sat.api.User(
+            role=SELECTED_ROLE,
+            admin=False,
+            login=gen_string('alphanumeric'),
+            password=password,
+            organization=[function_org],
+            location=[function_location],
+        ).create()
+        request.addfinalizer(user.delete)
+        user_cfg = user_nailgun_config(user.login, password)
+
+        hg_name = target_sat.api.HostGroup(
+            organization=[function_org],
+            location=[function_location],
+        ).create()
+        with target_sat.ui_session(user=user_cfg.auth[0], password=password) as session:
+            assert session.hostgroup.search(hg_name.name)[0]['Name'] == hg_name.name
+
```

Comment:
@shweta83 I think this new test is unnecessary and should be covered in the existing ones. Also, I see some tests set `admin=True` even for `non-admin` users, is that expected?

Reply:
All non-admin related test serves different purpose based on the roles assigned to it so I added this test with only specific roles assigned to the user. Please share any example where this can fit and I will update the existing test. Also, where do you see admin set to True?

Reply:
I think this is a valid scenario to check whether a non-admin user has access to the hostgroup in the same location and organization. However, @Shweta, I noticed that even though the non-admin user can locate the hostgroup, they cannot enter it to check for other sub-tasks within the hostgroup. This might be because they are a non-admin user.

@suggestion, you could add another scenario where an admin user navigates to the hostgroup and performs actions on it, allowing you to verify the permissions.

Reply:
I think this is a valid scenario to check whether a non-admin user has access to the hostgroup in the same location and organization. However, @Shweta, I noticed that even though the non-admin user can locate the hostgroup, they cannot enter it to check for other sub-tasks within the hostgroup. This might be because they are a non-admin user.

@suggestion, you could add another scenario where an admin user navigates to the hostgroup and performs actions on it, allowing you to verify the permissions.

Reply:
I think this scenario could be covered in `test_positive_create_delete_variable_with_overrides` https://github.com/SatelliteQE/robottelo/blob/ef6e48bc44a2dd808669ec43a4c0717a30181030/tests/foreman/ui/test_ansible.py#L34 
https://github.com/SatelliteQE/robottelo/blob/ef6e48bc44a2dd808669ec43a4c0717a30181030/tests/foreman/ui/test_ansible.py#L682 here `admin=True` is set for `non-admin` user.

Reply:
In any case, I leave it to you to decide if this scenario could be covered in the test I mentioned above.

Reply:
I would like to keep these scenarios separate.